tests/tools.py

```diff
@@ -193,7 +193,7 @@ def setUp(self):
 
     def tearDown(self):
         if not debug and self.tmp_dir is not None:
-            shutil.rmtree(self.tmp_dir)
+            shutil.rmtree(self.tmp_dir, ignore_errors=True)
```

Comment:
if we ignore errors and some files are not deleted subsequent tests that depend on the cleanup mail fail

Reply:
Fair enough, let me localize the cleanup to the flaky test.